pkg/vm/vm_test.go

```diff
@@ -3142,6 +3143,129 @@ func TestNegate(t *testing.T) {
 	assert.Equal(t, int64(-6), vm.estack.Pop().BigInt().Int64())
 }
 
+func TestCallE(t *testing.T) {
+	prog := []byte{byte(opcode.CALLE)}
+
+	//rvcount
+	prog = append(prog, byte(1))
+	//pcount
+	prog = append(prog, byte(0))
+
+	hash := util.Uint160{5, 2}
+	// script hash
+	prog = append(prog, hash.BytesBE()...)
+
+	prog = append(prog, byte(opcode.RET))
+
+	vm := load(prog)
+	vm.SetScriptGetter(func(in util.Uint160) ([]byte, bool) {
+		if in.Equals(hash) {
+			return makeProgram(opcode.DEPTH), true
+		}
+		return nil, false
+	})
+
+	vm.estack.PushVal(2)
+
+	runVM(t, vm)
+	elem := vm.estack.Pop() // depth should be 0, as it's a separate execution context
+
+	assert.Equal(t, int64(0), elem.BigInt().Int64())
+}
+
+func TestCallED(t *testing.T) {
+	prog := []byte{byte(opcode.CALLED)}
+
+	//rvcount
+	prog = append(prog, byte(1))
+	//pcount
+	prog = append(prog, byte(0))
+	prog = append(prog, byte(opcode.RET))
+
+	vm := load(prog)
+
+	vm.Context().hasDynamicInvoke = true
+
+	hash := util.Uint160{5, 2}
+	vm.SetScriptGetter(func(in util.Uint160) ([]byte, bool) {
```

Comment:
We certainly need to unify these tests in some table-driven fashion to avoid massive code duplication.

Reply:
Added unified structure for these tests:
```
struct {
	name             string
	hasDynamicInvoke bool
	hasScriptOnStack bool
	needSetRVCount   bool
	offset           byte //TODO: which type is offset?
	params           []interface{}
}
```
but have a question. Actually, we don't need these `hasDynamicInvoke` and `hasScriptOnStack` fields as far as we can evaluate this values in runtime. Should I remove them from testCase struct?